### DISCUSSION TITLE
how to change default timeout time for custom script?

### DISCUSSION THREAD

**Discussion by Muwusiz** [FormerLurker/Octolapse#793] [Q&A]
im currently running a dlsr with a script that saves it to the sd card instead of the pi. every time it takes a photo octolapse waits 30~ seconds for a photo to download to the pi so i end up having to wait 30 seconds in between each screenshot layer. and ideas on how to fix this?

**Comment by FormerLurker** [comment]
How long does it take for the image to be saved if you run your script from the command line?  Can you provide a screencap of the output?

The timeout is used to cancel a snapshot that is taking too long.  It doesn't control how long it actually takes to get a snapshot back.

**Reply by Muwusiz** [reply]
from the command line the script only takes around half a second to run.

here is the script.

```# Put the arguments sent by Octolapse into variables for easy use
SNAPSHOT_NUMBER=$1
DELAY_SECONDS=$2
DATA_DIRECTORY=$3
SNAPSHOT_DIRECTORY=$4
SNAPSHOT_FILENAME=$5
SNAPSHOT_FULL_PATH=$6

# IMPORTANT - You must add gphoto2 to your /etc/sudoers file in order to execute gphoto2 without sudo
# otherwise the following line will fail.
sudo gphoto2 --capture-image --keep

**Reply by FormerLurker** [reply]
But what happens if you run that script from the command line?  I'm particularly interested in the timing.

**Reply by FormerLurker** [reply]
Also, you shouldn't need the --keep flag.  Not sure what that would do, but --capture-image is supposed to keep the image on the SD, but not download it.

**Reply by Muwusiz** [reply]
okay. ill remove the --keep flag. what about the timing are you interested in? as i stated it only take around half a second to run

**Reply by FormerLurker** [reply]
> what about the timing are you interested in

Whoops, you answered, but I missed it somehow (it was too early).

If that's true, something is preventing Octolapse from receiving a return value from the script, and I'm not sure what that is.  Have you tried --trigger-capture instead of --capture-image?

I may need to run some tests on how --capture-image works.

**Reply by Muwusiz** [reply]
changing --capture-image to --trigger-capture fixes the 30 second delay!

**Reply by FormerLurker** [reply]
Awesome!  So what we've learned is that --capture-image is not returning, or is returning something different than what is expected.

I recommend running a quick test (test mode is great for this) to make sure there isn't any blurring, and that the snapshots are actually taking place with the bed stabilized.  Some cameras will return from trigger capture after the exposure, some before.  In any case, it's worth testing.

**Reply by FormerLurker** [reply]
Also, just curious, have you looked at [this guide](https://github.com/FormerLurker/Octolapse/wiki/V0.4---Configuring-a-DSLR-With-No-Download)?  That pretty much is what you are doing.  There's also a deferred download method that still renders a timelapse at the end by downloading all the images at once.  It's not compatible with all cameras, but worth a shot.  Finally, I'm curious if you're using Octolapse to render the images from your SD, or if you knew that was possible.  Let me know, because I rarely hear about those features/techniques.

**Reply by Muwusiz** [reply]
i have not seen that guide. ill make sure to check it out

im not using octolapse to render the timelapses because my camera doesn't support transferring files over usb. this is also why im saving them to the sd card instead of to the pi.
